### PR TITLE
feat(integer, fixpnt): migrate/implement parse() onto string_parse (Phase B1 of #835)

### PR DESCRIPTION
**Pull request by Ravenwater**
## Summary

- Migrates `integer::parse()` onto the shared `string_parse` foundation that landed in Phase A (#838) -- drops `std::regex` + `std::map`, fixes the previously-broken octal branch (was a `// TODO` returning false), and adds binary support.
- Implements `fixpnt::parse()` from scratch. **Discovery during this PR**: `fixpnt::parse()` was forward-declared in `fixpnt_fwd.hpp` and called by `operator>>` in `fixpnt_impl.hpp:2073`, but had no definition anywhere. A latent link error nobody had hit because no existing test exercises stream input on a fixpnt. This PR provides the definition.

Phase B1 explicitly excludes decimal-fraction parsing (`parse("3.14")` for fixpnt) -- that's the float-from-string problem and lands in Phase B2 alongside the analogous posit/cfloat work.

## Changes

| File | Change |
|------|--------|
| `include/sw/universal/number/integer/integer_impl.hpp` | Rewrites `parse()` body using `string_parse` primitives (~130 lines reduced to ~80, cleaner control flow). Implements binary + octal. Drops `<regex>` and `<map>` includes. |
| `include/sw/universal/number/fixpnt/fixpnt_impl.hpp` | Adds `parse()` definition (the dangling declaration finally has a body). Includes `<universal/utility/string_parse.hpp>`. |
| `static/integer/binary/api/string_parse.cpp` | **NEW** -- 33 assertions covering decimal/binary/octal/hex with sign handling and rejection of malformed inputs across integer<8,16,32,64>. |
| `static/fixpnt/binary/api/string_parse.cpp` | **NEW** -- 16 assertions covering bit-pattern fills (binary/octal/hex) and decimal-integer on fixpnt<8,4>, <9,4>, <16,8>. |

## Accepted syntax (both types)

```text
[+-]? ( 0[bB][01]+      |    // binary bit-pattern
        0[oO][0-7]+     |    // octal bit-pattern
        0[xX][0-9A-F']+ |    // hex bit-pattern (apostrophe allowed as separator)
        [0-9]+          )    // decimal integer
```

For fixpnt: bit-pattern parsers fill the underlying storage MSB-first (matches `setbits()`). Decimal parses as an integer and is stored at the rbits scale (`parse(\"5\")` on fixpnt<8,4> -> 5.0).

## Backward-compatibility notes

- **integer**: The previous octal branch was non-functional (`// TODO`). Real octal now requires the explicit `0o`/`0O` prefix (C-style leading-zero octal was never actually parsed).
- **integer**: Hex still accepts apostrophe as a digit separator (preserved from old behavior).
- **integer**: Decimal accepts a single optional leading `+` or `-` (previous regex permitted multiple sign chars -- a quirk; functional difference is nil).
- **fixpnt**: New function; no prior behavior to preserve.

## Test results

| Target | gcc build | gcc run | clang build | clang run |
|--------|-----------|---------|-------------|-----------|
| `bint_string_parse` | OK | 33/33 PASS | OK | 33/33 PASS |
| `fixpnt_string_parse` | OK | 16/16 PASS | OK | 16/16 PASS |
| `bint_api` (existing regression) | OK | PASS | (not re-checked) | -- |

## Phase B roadmap (this is B1 of B)

| Step | Scope | Status |
|------|-------|--------|
| **B1 (this PR)** | integer + fixpnt: bit-pattern + decimal-integer parsing via string_parse | **shipping** |
| B2 | Decimal-FP value parsing (`parse(\"3.14\")` -> closest representable). Targets fixpnt + posit + cfloat. Float-from-string. | future |
| B3 (likely) | posit + cfloat: migrate existing bit-pattern parse onto string_parse | future |

## Test plan

- [x] Fast CI passes (gcc + clang CI_LITE)
- [x] CodeRabbit feedback addressed
- [ ] Promote to ready when satisfied: \`gh pr ready <NN>\`

Relates to #835

Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Enhanced string parsing for fixed‑point and integer values: supports optional sign, base‑prefix detection (binary/octal/hex), decimal input, apostrophe digit separators, strict digit validation, and rejection of malformed/empty inputs.

* **Bug Fixes**
  * Stream extraction now sets failbit on parse failure and emits clearer diagnostics referencing fixpnt values.

* **Tests**
  * Added standalone test programs exercising parsing across formats and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/stillwater-sc/universal/pull/839?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: include/sw/universal/number/fixpnt/fixpnt_impl.hpp [hunk 1]
```diff
@@ -2065,13 +2066,115 @@ inline std::ostream& operator<<(std::ostream& ostr, const fixpnt<nbits, rbits, a
...
+// deferred to Phase B2 alongside posit/cfloat float-from-string.
+//
+// All scanning is delegated to the constexpr primitives in
+// `<universal/utility/string_parse.hpp>`.
+//
+// Returns true on successful parse, false on malformed input.
+template<unsigned nbits, unsigned rbits, bool arithmetic, typename bt>
+bool parse(const std::string& number, fixpnt<nbits, rbits, arithmetic, bt>& value) {
+	namespace sp = sw::universal::string_parse;
+	using FP = fixpnt<nbits, rbits, arithmetic, bt>;
```
Comment (Ravenwater) [inline include/sw/universal/number/fixpnt/fixpnt_impl.hpp]: don't use FP, as that is ambiguous and typically more associated with floating-point than with fixed-point. Instead of FP, use Fixpnt, that is clear and unambiguous.
